src/gg.regression.unity.bots/Editor/Scripts/CodeGenerators/CodeGeneratorUtils.cs

```diff
@@ -1,15 +1,26 @@
+using System.Linq;
 using System.Text.RegularExpressions;
 #if UNITY_EDITOR
+using System;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
 using UnityEditor;
+using UnityEngine;
 #endif
 
 namespace RegressionGames
 {
 #if UNITY_EDITOR
-    public class CodeGeneratorUtils
+    public static class CodeGeneratorUtils
     {
-        public static readonly string HeaderComment = $"/*\r\n* This file has been automatically generated. Do not modify.\r\n*/\r\n\r\n";
-        
+        public static readonly string HeaderComment = "/// <auto-generated>" + Environment.NewLine +
+                                                      "/// This file has been automatically generated. Do not modify." + Environment.NewLine +
+                                                      "/// </auto-generated>" + Environment.NewLine +
+                                                      "// ReSharper disable ClassNeverInstantiated.Global" + Environment.NewLine +
+                                                      "// ReSharper disable RedundantAssignment" + Environment.NewLine +
+                                                      "// ReSharper disable InconsistentNaming" + Environment.NewLine +
+                                                      "// ReSharper disable CheckNamespace" + Environment.NewLine  + Environment.NewLine;
```

Comment:
I'm on board with the `all` for these files.  We did a decent but not perfect job generating safe/reliable code.  It might not match every convention, and I'm sure it certainly won't in the future as conventions evolve in newer c# releases.

As for the name, I'm also good with moving the word generated to the 'end' of the name instead of the front where I have it now.


Current

`Generated_RGActions_<SourceBehaviourName>.cs`
`Generated_RGStateEntity_<SourceBehaviourName>.cs`

Proposed

`RGActions_<SourceBehaviourName>_Generated.cs`
`RGStateEntity_<SourceBehaviourName>_Generated.cs`

Reply:
updated to this new pattern and in all the samples

Reply:
Just a heads-up that if you want this to be detected by tools like Rider as generated code, the file name needs to end with `.Generated.cs` **not** `_Generated.cs`

<img width="325" alt="image" src="https://github.com/Regression-Games/RGUnityBots/assets/7574/06e0d3c0-97de-4e15-9d92-b7bdf958f168">

I'm still fine either way though, so this is non-blocking.

Reply:
Fair enough :)